[api-minor] Replace the `canvas` package with `@napi-rs/canvas`

Snuffleupagus opened the pull request:
The `@napi-rs/canvas` package has fewer dependencies, which should *hopefully* make installing and using it easier for `pdfjs-dist` end-users. (Over the years we've seen, repeatedly, that `canvas` can be difficult to install successfully.)
Furthermore, this package includes more functionality (such as `Path2D`) which reduces the overall number of dependencies in the PDF.js project.

One point to note is that `@napi-rs/canvas` is a fair bit newer than `canvas`, and has a lot fewer users, however looking at the commit history it does seem to be actively maintained.

Note that I've successfully tested the [Node.js examples](https://github.com/mozilla/pdf.js/tree/master/examples/node), in particular the `pdf2png` one, with this patch applied and things appear to work fine.

Please see:
 - https://www.npmjs.com/package/@napi-rs/canvas
 - https://github.com/Brooooooklyn/canvas

Snuffleupagus committed:
[api-minor] Replace the `canvas` package with `@napi-rs/canvas`

The `@napi-rs/canvas` package has fewer dependencies, which should *hopefully* make installing and using it easier for `pdfjs-dist` end-users. (Over the years we've seen, repeatedly, that `canvas` can be difficult to install successfully.)
Furthermore, this package includes more functionality (such as `Path2D`) which reduces the overall number of dependencies in the PDF.js project.

One point to note is that `@napi-rs/canvas` is a fair bit newer than `canvas`, and has a lot fewer users, however looking at the commit history it does seem to be actively maintained.

Note that I've successfully tested the [Node.js examples](https://github.com/mozilla/pdf.js/tree/master/examples/node), in particular the `pdf2png` one, with this patch applied and things appear to work fine.

Please see:
 - https://www.npmjs.com/package/@napi-rs/canvas
 - https://github.com/Brooooooklyn/canvas

nicolo-ribaudo commented on README.md:
```diff
@@ -57,9 +57,6 @@ all dependencies for PDF.js:
     $ npm install
 
-> [!NOTE]
-> On MacOS M1/M2 you may see some `node-gyp`-related errors when running `npm install`. This is because one of our dependencies, `"canvas"`, does not provide pre-built binaries for this platform and instead `npm` will try to build it from source. Please make sure to first install the necessary native dependencies using `brew`: https://github.com/Automattic/node-canvas#compiling.
```
Does the new package "just work" on MacOS? :)

Snuffleupagus replied:
Yes, it at least *installs* successfully there as shown in a modified CI run that I triggered previously: https://github.com/mozilla/pdf.js/actions/runs/11744596145

timvandermeij replied:
Note that `node-gyp` is not used by this new package, see the "About" section at https://github.com/Brooooooklyn/canvas, which is probably why this is not an issue anymore.

Snuffleupagus commented:
Before moving forward with this, the following points should probably be agreed upon:
 - Do we want to replace the `canvas` dependency?
 - Do we think that `@napi-rs/canvas` is a reasonable choice?
 - Test this sufficiently on "all" platforms (my testing has been on Windows).

Snuffleupagus committed:
Polyfill `ImageData` in Node.js environments

Given that `ImageData` has been supported for many years in all browsers, see [MDN](https://developer.mozilla.org/en-US/docs/Web/API/ImageData#browser_compatibility), we have a `typeof` check that's only necessary in Node.js environments.
Since the `@napi-rs/canvas` package provides that functionality, we can thus add an `ImageData` polyfill which allows us to ever so slightly simplify the code.

Snuffleupagus commented:
/botio test

calixteman commented:
I tested the patch on mac is the example and it works fine.

calixteman reviewed (approved):
LGTM. Thank you.

GreenFlag31 commented:
@Snuffleupagus 
I came across your [example](https://github.com/mozilla/pdf.js/blob/master/examples/node/pdf2png/pdf2png.mjs), but things are not working here. canvasFactory is a plain object, and does not possess a method `create()`
![image](https://github.com/user-attachments/assets/462782fa-5a2f-45fe-8474-a5e88021f1c1)
